### PR TITLE
🌱 Bump golang to 1.22 in release 0.6

### LINKED ISSUE #1803: Bump BMO 0.6 release branch to Go 1.22
Release branch release-0.6 has Golang 1.21.x in use. Golang 1.21 support ends in August 2024, when Golang 1.23 is released.

BMO 0.6 is in support for quite a while still, so we need to bump it to Go 1.22 before next patch release.

### PR DESCRIPTION
**Pull request by NymanRobin**
This is done since golang 1.21 is deprecating in August 2024

Fixes #1803

### CODE REVIEW

**Comment by NymanRobin** [comment]
Here is the corresponding project-infra change: https://github.com/metal3-io/project-infra/pull/823
That will be needed before this will pass the tests

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/metal3-io/baremetal-operator
 
-go 1.21
+go 1.22
```
Comment (dtantsur) [inline go.mod:3]: This is not necessary unless a dependency is forcing us to.

**Review by dtantsur — CHANGES_REQUESTED** [review]
Still is my regular request to avoid unnecessary changes to go.mod, especially on stable branches.

File: go.mod (hunk 1 above)
Comment (NymanRobin) [reply go.mod]: Sure, I see the concern of unnecessary changes on stable branches, should be fixed now!

**Comment by tuminoid** [comment]
@dtantsur I think your concern was fixed, can you remove the "request changes" review, as it is blocking merging?

**Pull request merged**